### DISCUSSION TITLE
Classes randomly getting purged with Tailwind, Next.js when deploying to production (vercel)

### DISCUSSION THREAD

**Discussion by iNishant** [tailwindlabs/tailwindcss#4299] [Help]
Some of the tailwind classes are randomly get purged when deploying to production (vercel). Everything works perfectly locally. In the below example, the class `mt-40` randomly gets purged in prod.

Any idea why this might be happening?

Versions: "tailwindcss": "^2.1.2",  "next": "^10.2.0"

![Screenshot 2021-05-10 at 5 49 35 PM](https://user-images.githubusercontent.com/10304344/117658317-37036400-b1b8-11eb-872a-63c4b2153be0.png)

![Screenshot 2021-05-10 at 5 49 57 PM](https://user-images.githubusercontent.com/10304344/117658327-39fe5480-b1b8-11eb-8a4d-0a4e5fbddbb3.png)

**Comment by simonswiss** [comment]
Hey!

Are you "constructing" that margin-top class dynamically, based on a prop? As in...

```js
<div className={`text-3xl mt-${props.offset} ...`}>
```

?

That won't be purgeable, as PurgeCSS needs to see the entire class name string to make the decision to keep it in the generated CSS.

Another potential issue could be that you haven't set the paths correctly in your `purge` config options.

Are you able to share a minimal repro of the issue?

**Reply by iNishant** [reply]
No I am not constructing. My paths in purge are also correct. Will share repro soon.

**Comment by iNishant** [comment]
@simonswiss can I invite you to the private repo? Its a single page next.js application
